homeassistant/components/miele/services.py

```diff
@@ -0,0 +1,76 @@
+"""Services for Miele integration."""
+
+import logging
+from typing import cast
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.const import ATTR_DEVICE_ID
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.service import async_extract_config_entry_ids
+
+from .const import DOMAIN
+from .coordinator import MieleConfigEntry
+
+ATTR_PROGRAM_ID = "program_id"
+ATTR_DURATION = "duration"
+
+SERVICE_PROGRAM = vol.Schema(
```

Comment:
```suggestion
SERVICE_SET_PROGRAM = "set_program"
SERVICE_SET_PROGRAM_SCHEMA = vol.Schema(
```
Easier to reuse in tests and such :)

Also, why is it `set_program`? Do we set a program? Or do we start it?

Reply:
Depending on Miele's implementation for different products/models. Setup of program is common for all. Some models will auto-start and some models needs a start command (e.g. button press). No documentation - as usual, but the users can easily find out how their specific appliance behaves.